### PR TITLE
[1.20.1] Fix Slot Index for Inventory Ticking Items

### PR DESCRIPTION
**Pull request by Spectrall368**
This PR fixes the slot index for inventory ticking items, similar to: #9736

### CODE REVIEW

File: src/main/java/net/minecraftforge/common/extensions/IForgeItem.java [hunk 1]
```diff
@@ -341,14 +341,15 @@ default void onArmorTick(ItemStack stack, Level level, Player player)
...
-    	}
+        var inv = player.getInventory();
+        int vanillaIndex = slotIndex;
+        if (slotIndex >= inv.items.size()) {
+            vanillaIndex -= inv.items.size();
+            if (vanillaIndex >= inv.armor.size())
+                vanillaIndex -= inv.armor.size();
+            else
+                onArmorTick(stack, level, player);
+        }
```
Comment (Jonathing) [inline src/main/java/net/minecraftforge/common/extensions/IForgeItem.java]: Please preserve the formatting that was used prior.

**Review by Jonathing — CHANGES_REQUESTED** [review]

File: src/main/java/net/minecraftforge/common/extensions/IForgeItem.java (hunk 1 above)
Comment (Spectrall368) [reply src/main/java/net/minecraftforge/common/extensions/IForgeItem.java]: Ok

Comment (Spectrall368) [reply src/main/java/net/minecraftforge/common/extensions/IForgeItem.java]: Done

**Comment by Jonathing** [comment]
If you are unsure that your PR is ready for the project, please mark it as a draft.

**Comment by Spectrall368** [comment]
Since I don't know how to resolve the error to make all the checks work, I'm closing this PR

**Pull request closed without merging**